core/src/main/java/org/opensearch/sql/ast/AbstractNodeVisitor.java

```diff
@@ -256,7 +257,11 @@ public T visitAD(AD node, C context) {
     return visitChildren(node, context);
   }
 
-  public T visitHighlight(HighlightFunction node, C context) {
+  public T visitHighlightFunction(HighlightFunction node, C context) {
+    return visitChildren(node, context);
+  }
+
+  public T visitHighlight(Highlight node, C context) {
     return visitChildren(node, context);
   }
```

Comment:
What is the difference?

Reply:
`visitHighlightFunction` takes a `HighlightFunction` node and has an overriding function in `ExpressionAnalyzer`, while `visitHighlight` takes a `highlight` node and has an overriding function in `Analyzer`. This is because for `PPL` I was unable to resolve a highlight field as an expression like I did in `SQL`. Instead I use `HighlightOperator` and use the `Analyzer` to get around this issue. This issue arose because of the way the parser was implemented, the function `highlight` syntax of not having `highlight` in a `fields` command, and the parser implementation of `AllFields` without a fields command.

We may want to consider updating SQL to use the `highlightOperator` rather than `HighlightExpression`...